app/assessmentFromCrn/post.controller.js

```diff
@@ -1,30 +1,48 @@
 // const { logger } = require('../../common/logging/logger')
 const { assessmentSupervision } = require('../../common/data/hmppsAssessmentApi')
 
-const startAssessmentFromCrn = ({ params: { crn, deliusEventId, assessmentType }, user }, res) => {
-  return startAssessment(crn, deliusEventId, assessmentType, user?.token, user?.id, res)
-}
+const getErrorMessageFor = (status, offenderDetails, user) => {
+  // We should use an error enum here, status codes may be a bit ambiguous
+  if (status === 401) {
+    return 'You do not have permission to complete this type of assessment. Speak to your manager and ask them to request a change to your level of authorisation.'
+  }
+  if (status === 400) {
+    return `${offenderDetails.name} is showing as a possible duplicate record under ${user.areaName} PNC ${offenderDetails.pnc} Log into OASys to manage the duplication. If you need help, contact the OASys Application Support team`
+  }
```

Comment:
the offender-assessments-api is returning 409 for duplicate offender 

```
@ExceptionHandler(DuplicateOffenderRecordException::class)
  fun handle(e: DuplicateOffenderRecordException): ResponseEntity<ErrorResponse?> {
    log.error("DuplicateOffenderRecordException: {}", e.message)
    return ResponseEntity(ErrorResponse(status = 409, developerMessage = e.message, userMessage = e.message), HttpStatus.CONFLICT)
  }
```

Not sure if this error is coming from hmpps-asessments-api instead? and transforming the duplicate offender 409 to bad request 400?

Reply:
Just checked that's correct the `hmpps-assessments-api` translate 409-400 

```
@ExceptionHandler(DuplicateOffenderRecordException::class)
  @ResponseStatus(HttpStatus.BAD_REQUEST)
  fun handle(e: DuplicateOffenderRecordException): ResponseEntity<ErrorResponse?> {
    log.error("DuplicateOffenderRecordException: ${e.message} with extra information ${e.extraInfoMessage}")
    return ResponseEntity(ErrorResponse(status = 400, developerMessage = e.message), HttpStatus.BAD_REQUEST)
  }
```

Reply:
Yup 💯  - the HMPPS Assessments API is swallowing the 409 and returning a 400, I've created a PR on the API to pass a reason enum to the frontend to for handling of errors, status codes alone seemed too ambiguous